src/devices/bus/heathzenith/h8/h8bus.cpp

```diff
@@ -0,0 +1,242 @@
+	device_t(mconfig, type, tag, owner, clock),
+	device_single_card_slot_interface(mconfig, *this),
+	m_h8bus(*this, finder_base::DUMMY_TAG),
+	m_h8bus_slottag(nullptr)
+{
+}
+
+void h8bus_slot_device::device_start()
+{
+}
+
+void h8bus_slot_device::device_resolve_objects()
+{
+	device_h8bus_card_interface *dev = get_card_device();
+
+	if (dev)
+	{
+		dev->set_h8bus_tag(m_h8bus.target(), m_h8bus_slottag);
+		m_h8bus->add_h8bus_card(*dev);
+	}
+}
+
+
+DEFINE_DEVICE_TYPE(H8BUS, h8bus_device, "h8bus", "Heathkit H8 bus (Benton Harbor Bus)")
+
+h8bus_device::h8bus_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	h8bus_device(mconfig, H8BUS, tag, owner, clock)
+{
+}
+
+h8bus_device::h8bus_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, type, tag, owner, clock),
+	device_memory_interface(mconfig, *this),
+	m_mem_config("mem", ENDIANNESS_LITTLE, 16, 16, 0, address_map_constructor(FUNC(h8bus_device::mem_map), this)),
+	m_io_config("pio", ENDIANNESS_LITTLE, 8, 8, 0, address_map_constructor(FUNC(h8bus_device::io_map), this)),
+	m_p1_card(nullptr),
+	m_p2_card(nullptr)
+{
+}
+
+h8bus_device::~h8bus_device()
+{
+}
+
+void h8bus_device::interface_pre_start()
+{
+	// Verify required cards are installed.
+	if (!m_p1_card)
+	{
+		fatalerror("Can't find H-8 P1 - Front panel card\n");
+	}
+
+	if (!m_p2_card)
+	{
+		fatalerror("Can't find H-8 P2 - CPU card\n");
+	}
+
+}
+
+void h8bus_device::device_start()
+{
+}
+
+void h8bus_device::device_reset()
+{
+}
+
+void h8bus_device::mem_map(address_map &map)
+{
+	map.unmap_value_high();
+}
+
+void h8bus_device::io_map(address_map &map)
+{
+	map.unmap_value_high();
+	map.global_mask(0xff);
+}
+
+device_memory_interface::space_config_vector h8bus_device::memory_space_config() const
+{
+	return space_config_vector { std::make_pair(AS_PROGRAM, &m_mem_config), std::make_pair(AS_IO, &m_io_config) };
+}
+
+void h8bus_device::add_h8bus_card(device_h8bus_card_interface &card)
+{
+	m_device_list.emplace_back(card);
+	if (strcmp(card.m_h8bus_slottag, "p1") == 0)
+	{
+		m_p1_card = (device_h8bus_p1_card_interface*) &card;
+	}
+	else if (strcmp(card.m_h8bus_slottag, "p2") == 0)
+	{
+		m_p2_card = (device_h8bus_p2_card_interface*) &card;
+	}
+}
+
+void h8bus_device::set_int1_line(int state)
+{
+	m_p2_card->int1_w(state);
+}
+
+void h8bus_device::set_int2_line(int state)
+{
+	m_p2_card->int2_w(state);
+}
+
+void h8bus_device::set_int3_line(int state)
+{
+	m_p2_card->int3_w(state);
+}
+
+void h8bus_device::set_int4_line(int state)
+{
+	m_p2_card->int4_w(state);
+}
+
+void h8bus_device::set_int5_line(int state)
+{
+	m_p2_card->int5_w(state);
+}
+
+void h8bus_device::set_int6_line(int state)
+{
+	m_p2_card->int6_w(state);
+}
+
+void h8bus_device::set_int7_line(int state)
+{
+	m_p2_card->int7_w(state);
+}
```

Comment:
Remember any card on the bus can see when another card asserts one of these, as they’re all wired together.  It’s better to use an approach that allows telling all the cards the state they’d see on these lines.